src/pages/en/guides/upgrade-to/v2.mdx

```diff
@@ -0,0 +1,359 @@
+
+
+##### What should I do?
+
+Remove `extendDefaultPlugins: true` in your configuration. This is now Astro's default behavior in 2.0, and you can delete this line without any replacement.
+
+```js del={6}
+// astro.config.mjs
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  markdown: {
+    extendDefaultPlugins: true,
+  }
+});
+```
+
+#### Added: `gfm` and `smartypants`
+
+Astro 2.0 replaces `markdown.extendDefaultPlugins` with separate Boolean options to control each of Astro's built-in default Markdown plugins individually. These are enabled by default, and can be set to `false` to remove each plugin independently.
+
+##### What should I do?
+
+Replace `extendDefaultPlugins: false` and add the flags to disable each plugin individually instead.
+
+- `markdown.gfm: false` disables GitHub-Flavored Markdown
+- `markdown.smartypants: false` disables SmartyPants
+
+```js del={6} ins={7-8}
+// astro.config.mjs
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  markdown: {
+    extendDefaultPlugins: false,
+    smartypants: false,
+    gfm: false,
+  }
+});
+```
+
+### Changed: MDX plugin configuration
+
+#### Replaced: `extendPlugins` -> `extendMarkdownConfig`
+
+In 1.0, `.mdx.extendPlugins` managed how your MDX files should inherit your Markdown plugin configuration. This setting was used to indicate whether all Markdown plugins (`markdown`), or only Astro's default plugins (`defaults`) should be applied to MDX.
+
+Astro 2.0 replaces `mdx.extendPlugins` with three new, independently-configurable options:
+
+- **[`mdx.extendMarkdownConfig`](/en/guides/integrations-guide/mdx/#extendmarkdownconfig)** (`true` by default) to inherit all of your Markdown configuration or none of it.
+- **`mdx.gfm`** (`true` by default) to enable or disable GitHub-Flavored Markdown in MDX.
+- **`mdx.smartypants`** (`true` by default) to enable or disable SmartyPants in MDX.
+
+ ##### What should I do?
+
+Delete `extendPlugins: 'markdown'` in your configuration. This is now default behavior. 
+
+```js del={6}
+// astro.config.mjs
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  mdx: {
+    extendPlugins: markdown,
+  }
+});
+```
+
+Replace `extendPlugins: 'defaults'` with `extendMarkdownConfig: false` and add the [separate options for GitHub-Flavored Markdown and SmartyPants](/en/guides/integrations-guide/mdx/#options-inherited-from-markdown-config) to enable these default plugins individually in MDX.
+
+```js del={6} ins={7-9}
+// astro.config.mjs
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  mdx: {
+    extendPlugins: defaults,
+    extendMarkdownConfig: false,
+    smartypants: true,
+    gfm: true,
+  }
+});
+```
```

Comment:
These options are passed to the MDX integration not to an `mdx` config key, so reworked this to reflect that:

````suggestion
In 1.0, the MDX integration’s `extendPlugins` option managed how your MDX files should inherit your Markdown configuration.

Astro 2.0 replaces `extendPlugins` with three new, independently-configurable options:

- **[`extendMarkdownConfig`](/en/guides/integrations-guide/mdx/#extendmarkdownconfig)** (`true` by default) to inherit all of your Markdown configuration or none of it.
- **`gfm`** (`true` by default) to enable or disable GitHub-Flavored Markdown in MDX.
- **`smartypants`** (`true` by default) to enable or disable SmartyPants in MDX.

##### What should I do?

Delete `extendPlugins: 'markdown'` in your configuration. This is now default behavior. 

```js del={8}
// astro.config.mjs
import { defineConfig } from 'astro/config';
import mdx from '@astrojs/mdx';

export default defineConfig({
  integrations: [
    mdx({
      extendPlugins: 'markdown',
    }),
  ],
});
```

Replace `extendPlugins: 'defaults'` with `extendMarkdownConfig: false` and add the [separate options for GitHub-Flavored Markdown and SmartyPants](/en/guides/integrations-guide/mdx/#options-inherited-from-markdown-config) to enable these default plugins individually in MDX.

```js del={8} ins={9-11}
// astro.config.mjs
import { defineConfig } from 'astro/config';
import mdx from '@astrojs/mdx';

export default defineConfig({
  integrations: [
    mdx({
      extendPlugins: 'defaults',
      extendMarkdownConfig: false,
      smartypants: true,
      gfm: true,
    }),
  ],
});
```
````